syntax-highlighting-code-block.php

```diff
@@ -222,3 +259,62 @@ function render_block( $attributes, $content ) {
 		);
 	}
 }
+
+/**
+ * Initialize admin settings.
+ */
+function admin_init() {
+	register_setting( 'syntax_highlighting', OPTION_NAME );
+}
+add_action( 'admin_init', __NAMESPACE__ . '\admin_init' );
+
+/**
+ * Sanitize the given theme name to make sure it's a valid highlight.js theme.
+ *
+ * @param string $input Value of the dropdown.
+ *
+ * @return string The sanitized theme name
+ */
+function sanitize_theme_name( $input ) {
+	require_once __DIR__ . '/vendor/scrivo/highlight.php/HighlightUtilities/functions.php';
+
+	$themes = \HighlightUtilities\getAvailableStyleSheets();
+
+	return in_array( $input, $themes, true ) ? sanitize_text_field( $input ) : DEFAULT_THEME;
```

Comment:
If `$input` is already in `$themes`, then isn't `sanitize_text_field()` somewhat redundant? Perhaps using a `sanitize_callback` is not right in this case, and a `validate_callback` should be used instead. This allows you to just return an error instead of forcibly reverting the theme to the default.

Read more at https://make.wordpress.org/core/2016/07/05/customizer-apis-in-4-6-for-setting-validation-and-notifications/

So then the `validate_callback` could return with:

```php
function validate_theme_name( $validity, $input ) {
	require_once __DIR__ . '/vendor/scrivo/highlight.php/HighlightUtilities/functions.php';
	$themes = \HighlightUtilities\getAvailableStyleSheets();
	if ( in_array( $input, $themes, true ) ) {
		$validity->add( 'invalid_theme', __( 'Unrecognized theme', 'syntax-highlighting-code-block' ) );
	}
	return $validity;
}
```

Reply:
Ok, this Customizer API is pretty freaking neat! Moved to a validator function instead of a sanitizer.